ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/product/Product.java

```diff
@@ -681,6 +686,25 @@ public void setUseFeatures(boolean use) {
 		}
 	}
 
+	@Override
+	public boolean includeRequirementsAutomatically() {
+		return fAutoIncludeRequirements;
+	}
+
+	@Override
+	public void setIncludeRequirementsAutomatically(boolean includeRequirements) {
+		boolean old = fAutoIncludeRequirements;
+		fAutoIncludeRequirements = includeRequirements;
+		if (isEditable()) {
+			firePropertyChanged(P_INCLUDE_REQUIREMENTS_AUTOMATICALLY, Boolean.toString(old),
+					Boolean.toString(fAutoIncludeRequirements));
+		}
+	}
+
+	private boolean getAutoAddRequirementsDefault() {
+		return fUseFeatures; // true if features are used, else false
```

Comment:
in my experience, tycho does not include requirements for feature based products

Reply:
My experiments showed the opposite: Tycho includes requirements for feature-based products as well as for plugin-based products. But I have not yet followed it in the P2 source-code.